src/Agent.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Laravel\Jetstream;
+
+use Detection\MobileDetect;
+
+/**
+ * @copyright Originally created by Jens Segers: https://github.com/jenssegers/agent
+ */
+class Agent extends MobileDetect
+{
+    /**
+     * List of additional operating systems.
+     *
+     * @var array<string, string>
+     */
+    protected static $additionalOperatingSystems = [
+        'Windows' => 'Windows',
+        'Windows NT' => 'Windows NT',
+        'OS X' => 'Mac OS X',
+        'Debian' => 'Debian',
+        'Ubuntu' => 'Ubuntu',
+        'Macintosh' => 'PPC',
+        'OpenBSD' => 'OpenBSD',
+        'Linux' => 'Linux',
+        'ChromeOS' => 'CrOS',
+    ];
+
+    /**
+     * List of additional browsers.
+     *
+     * @var array<string, string>
+     */
+    protected static $additionalBrowsers = [
+        'Opera Mini' => 'Opera Mini',
+        'Opera' => 'Opera|OPR',
+        'Edge' => 'Edge|Edg',
+        'Coc Coc' => 'coc_coc_browser',
+        'UCBrowser' => 'UCBrowser',
+        'Vivaldi' => 'Vivaldi',
+        'Chrome' => 'Chrome',
+        'Firefox' => 'Firefox',
+        'Safari' => 'Safari',
+        'IE' => 'MSIE|IEMobile|MSIEMobile|Trident/[.0-9]+',
+        'Netscape' => 'Netscape',
+        'Mozilla' => 'Mozilla',
+        'WeChat' => 'MicroMessenger',
+    ];
+
+    /**
+     * Get the platform name from User Agent.
+     *
+     * @return string|null
+     */
+    public function platform()
+    {
+        return $this->findDetectionRulesAgainstUserAgent(
+            $this->mergeRules(MobileDetect::getOperatingSystems(), static::$additionalOperatingSystems)
+        );
+    }
+
+    /**
+     * Get the browser name from User Agent.
+     *
+     * @return string|null
+     */
+    public function browser()
+    {
+        return $this->findDetectionRulesAgainstUserAgent(
+            $this->mergeRules(static::$additionalBrowsers, MobileDetect::getBrowsers())
+        );
+    }
+
+    /**
+     * Check if the device is a desktop computer.
+     *
+     * @return bool
+     */
+    public function isDesktop()
+    {
+        // Check specifically for cloudfront headers if the useragent === 'Amazon CloudFront'
+        if (
+            $this->getUserAgent() === static::$cloudFrontUA
+            && $this->getHttpHeader('HTTP_CLOUDFRONT_IS_DESKTOP_VIEWER') === 'true'
+        ) {
```

Comment:
This requires https://github.com/serbanghita/Mobile-Detect/blob/b73a5bdbcec74a6d33389cc8fe3a5af5d0965d01/src/MobileDetect.php#L1036-L1038 to actually work.